packages/core/src/connectivity/scp-cf/destination/destination-from-service.ts

```diff
@@ -156,30 +159,33 @@ class DestinationFromServiceRetriever {
   }
 
   readonly decodedProviderClientCredentialsToken: JwtPayload;
-  private options: DestinationOptions;
+  private options: RequiredProperties<
+    DestinationOptions,
+    'isolationStrategy' | 'selectionStrategy' | 'useCache'
+  >;
 
   private constructor(
     readonly name: string,
     options: DestinationOptions,
     readonly decodedUserJwt: JwtPayload | undefined,
     readonly providerClientCredentialsToken: string
   ) {
-    this.options = { ...options };
     this.decodedProviderClientCredentialsToken = decodeJwt(
       providerClientCredentialsToken
     );
 
-    this.options.isolationStrategy =
-      options.isolationStrategy || IsolationStrategy.Tenant;
-    this.options.selectionStrategy =
-      options.selectionStrategy || subscriberFirst;
-    this.options.useCache = options.useCache || false;
+    this.options = {
+      isolationStrategy: IsolationStrategy.Tenant,
+      selectionStrategy: subscriberFirst,
+      useCache: false,
+      ...options
+    };
```

Comment:
👍  I like this, but it is a bit implicit that `...options` may overrides those default values. Maybe it would be better like this?
```suggestion
    const defaultOptions =  {
      isolationStrategy: IsolationStrategy.Tenant,
      selectionStrategy: subscriberFirst,
      useCache: false,
      ...options
    };
    this.options = { ...defaultOptions, ...options };
```